tests/test_highlighting/highlighter.py

```diff
@@ -0,0 +1,157 @@
+
+#######################################################################################
+##                     This part concerns the simulation by Ace                      ##
+#######################################################################################
+
+
+import re
+
+# Transcription of the used tokens into a symbol (a letter) in order to apply a coloring
+Tokencode = {
+    "text" :               'T',
+    "keyword" :            'K',
+    "comment" :            'C',
+    "variable" :           'N',
+    "constant.character" : 'S',
+    "invalid" :            'I',
```

Comment:
This is clever! I like the idea! I wonder if it's going to be annoying to edit (don't forget that tests also need to evolve over time as the code base changes, they're not set in stone, and readability and maintainability count a lot for tests as well!), but the encoding can always be changed.

A potential alternative encoding could be something like:

```
[print:kw] hello world
```

Which would have the advantage of being more readable and you don't have to fiddle so much with the exact character offsets. Something to think about.